OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -605,8 +605,10 @@ public void ResolveOrder(Actor self, Order order)
 				{
 					if (IsPlane)
 						self.QueueActivity(new ReturnToBase(self, Info.AbortOnResupply));
-					else
+					else if (!Info.RepairBuildings.Contains(order.TargetActor.Info.Name))
```

Comment:
Another reason to keep the check in Aircraft: Currently ReturnToBase and HeliReturnBase don't receive here Actor dest parameter so they don't know the original target and can't decide to do nothing for Repair order. We could pass it, but it could have side effects, is more code and new hacks. I propose to either accept this bug fix in Aircraft.ResolveOrder or close this PR.